conans/client/installer.py

```diff
@@ -311,6 +317,41 @@ def _node_concurrently_installed(self, node, package_folder):
             if node.update_manifest == read_manifest:
                 return True
 
+    def _load_editables_cpp_info(self):
+        editables_path = self._client_cache.default_editable_path
+        if os.path.exists(editables_path):
+            return EditableCppInfo.create(editables_path, require_namespace=True)
+        return None
+
+    def _handle_node_editable(self, node):
+        # Get source of information
+        package_layout = self._client_cache.package_layout(node.conan_ref)
+        base_path = package_layout.conan()
+        self._call_package_info(node.conanfile, package_folder=base_path)
+
+        # Try with package-provided file
+        package_layout_file = package_layout.editable_package_layout_file()
+        if os.path.exists(package_layout_file):
+            editable_cpp_info = EditableCppInfo.create(package_layout_file,
+                                                       require_namespace=False)
+            editable_cpp_info.apply_to(node.conanfile.name,
+                                       node.conanfile.cpp_info,
+                                       base_path=base_path,
+                                       settings=node.conanfile.settings,
+                                       options=node.conanfile.options)
+
+        # Try with the profile-like file
+        elif self._editable_cpp_info and self._editable_cpp_info.has_info_for(node.conanfile.name):
+            self._editable_cpp_info.apply_to(node.conanfile.name,
+                                             node.conanfile.cpp_info,
+                                             base_path=base_path,
+                                             settings=node.conanfile.settings,
+                                             options=node.conanfile.options)
+
+        # Use `package_info()` data
+        else:
+            pass  # TODO: Use `package_info()` data
```

Comment:
I think this is not a TODO, it is already done. If defined above the ``package_folder=base_path`` pointing to the local folder, it should work out of the box.

Reply:
You are right. It is working right now, I've added a test case for that.